jib-core/src/main/java/com/google/cloud/tools/jib/hash/DigestUtil.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2017 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.hash;
+
+import com.google.cloud.tools.jib.blob.BlobDescriptor;
+import com.google.cloud.tools.jib.image.DescriptorDigest;
+import com.google.cloud.tools.jib.json.JsonTemplate;
+import com.google.cloud.tools.jib.json.JsonTemplateMapper;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+/**
+ * Utility class for computing a digest for various inputs while optionally writing to an output
+ * stream.
+ */
+// Note: intentionally this class does not depend on Blob, as Blob classes depend on this class.
+// TODO: BlobDescriptor is merely a tuple of (size, digest). Rename BlobDescriptor to something
+// more general.
+public class DigestUtil {
```

Comment:
As mentioned in the meeting, we should maybe avoid naming with "Util", so maybe this class could just be called `Digests` or something?